gammapy/image/models/new.py

```diff
@@ -51,32 +66,214 @@ class SkyGaussian2D(SkySpatialModel):
+
+class SkyDisk2D(SkySpatialModel):
+    r"""Constant radial disk model.
+
+    .. math::
+
+        \phi(lon, lat) = \frac{1}{2 \pi (1 - \cos{r}) } \cdot
+                \begin{cases}
+                    1 & \text{for } \theta \leq r_0 \\
+                    0 & \text{for } \theta < r_0
+                \end{cases}
+
+    where :math:`\theta` is the sky separation
+
+    Parameters
+    ----------
+    lon_0: `~astropy.coordiantes.Longitude`
+        :math:`lon_0`
+    lat_0: `~astropy.coordinates.Latitude`
+        :math:`lat_0`
+    r_0: `~astropy.coordinates.Angle`
+        :math:`r_0`
+    """
+
+    def __init__(self, lon_0, lat_0, r_0):
+        self.parameters = ParameterList([
+            Parameter('lon_0', Longitude(lon_0)),
+            Parameter('lat_0', Latitude(lat_0)),
+            Parameter('r_0', Angle(r_0))
+        ])
+
+    @staticmethod
+    def evaluate(lon, lat, lon_0, lat_0, r_0):
+        """Evaluate the model (static function)."""
+        sep = angular_separation(lon, lat, lon_0, lat_0)
+        norm = 1. / (2 * np.pi * (1 - np.cos(r_0)))
+        val = norm if sep <= r_0 else 0
+        return val / u.deg ** 2
+
+
+class SkyShell2D(SkySpatialModel):
+    r"""Shell model
+
+    .. math::
+
+        \phi(lon, lat) = \frac{3}{2 \pi (r_o^3 - r_i^3)} \cdot
+                \begin{cases}
+                    \sqrt{r_o^2 - \theta^2} - \sqrt{r_i^2 - \theta^2} &
+                                 \text{for } \theta \lt r_i \\
+                    \sqrt{r_o^2 - \theta^2} & 
+                                 \text{for } r_i \leq \theta \lt r_o \\
+                    0 & \text{for} \theta > r_o
+                \end{cases}
+
+    where :math:`\theta` is the sky separation. Note that the normalization is
+    a small angle approximation and gives correct results only up to radii of
+    30 deg.
+
+    Parameters
+    ----------
+    lon_0: `~astropy.coordiantes.Longitude`
+        :math:`lon_0`
+    lat_0: `~astropy.coordinates.Latitude`
+        :math:`lat_0`
+    r_i: `~astropy.coordinates.Angle`
+        :math:`r_i`
+    r_o: `~astropy.coordinates.Angle`
+        :math:`r_o`
+    """
+    def __init__(self, lon_0, lat_0, r_i, r_o):
+        self.parameters = ParameterList([
+            Parameter('lon_0', Longitude(lon_0)),
+            Parameter('lat_0', Latitude(lat_0)),
+            Parameter('r_i', Angle(r_i)),
+            Parameter('r_o', Angle(r_o))
+        ])
+
+    @staticmethod
+    def evaluate(lon, lat, lon_0, lat_0, r_i, r_o):
+        """Evaluate the model (static function)."""
+        sep = angular_separation(lon, lat, lon_0, lat_0)
+        term1 = np.sqrt(r_o**2-sep**2)
+        term2 = term1 - np.sqrt(r_i**2-sep**2)
+        norm = 3 / (2 * np.pi * (r_o ** 3 - r_i ** 3))
+
+        if sep < r_o:
+            if sep < r_i:
+                val = term2
+            else:
+                val = term1
+        else:
+            val = 0
+
+        return norm * val 
+
+
+class SkyTemplate2D(SkySpatialModel):
```

Comment:
Note that this is a copy&paste (+ adaptions) from the old model classes, I'm nore sure if we want to keep it in this form